Description: Add a system-wide preferences folder
Author: Alexander Sack <asac@ubuntu.com>
Forwarded: no

Index: firefox-trunk-17.0~a1~hg20120807r101567/toolkit/xre/nsXREDirProvider.cpp
===================================================================
--- firefox-trunk-17.0~a1~hg20120807r101567.orig/toolkit/xre/nsXREDirProvider.cpp	2012-08-07 03:12:55.000000000 +0100
+++ firefox-trunk-17.0~a1~hg20120807r101567/toolkit/xre/nsXREDirProvider.cpp	2012-08-07 12:10:15.303809140 +0100
@@ -613,6 +613,7 @@
 }
 
 static const char *const kAppendPrefDir[] = { "defaults", "preferences", nullptr };
+static const char *const kAppendSysPrefDir[] = { "defaults", "syspref", nullptr };
 
 #ifdef DEBUG_bsmedberg
 static void
@@ -654,6 +655,9 @@
     LoadDirIntoArray(mXULAppDir, kAppendPrefDir, directories);
     LoadDirsIntoArray(mAppBundleDirectories,
                       kAppendPrefDir, directories);
+    LoadDirIntoArray(mXULAppDir, kAppendSysPrefDir, directories);
+    LoadDirsIntoArray(mAppBundleDirectories,
+                      kAppendSysPrefDir, directories);
 
     rv = NS_NewArrayEnumerator(aResult, directories);
   }
Index: firefox-trunk-17.0~a1~hg20120807r101567/xpcom/io/nsAppDirectoryServiceDefs.h
===================================================================
--- firefox-trunk-17.0~a1~hg20120807r101567.orig/xpcom/io/nsAppDirectoryServiceDefs.h	2012-08-07 03:13:18.000000000 +0100
+++ firefox-trunk-17.0~a1~hg20120807r101567/xpcom/io/nsAppDirectoryServiceDefs.h	2012-08-07 11:38:13.751730203 +0100
@@ -30,6 +30,7 @@
 
 #define NS_APP_DEFAULTS_50_DIR                  "DefRt"         // The root dir of all defaults dirs
 #define NS_APP_PREF_DEFAULTS_50_DIR             "PrfDef"
+#define NS_APP_SYSPREF_DEFAULTS_50_DIR          "SysPrfDef"
 #define NS_APP_PROFILE_DEFAULTS_50_DIR          "profDef"       // The profile defaults of the "current"
                                                                 // locale. Should be first choice.
 #define NS_APP_PROFILE_DEFAULTS_NLOC_50_DIR     "ProfDefNoLoc"  // The profile defaults of the "default"
Index: firefox-trunk-17.0~a1~hg20120807r101567/xpcom/io/nsAppFileLocationProvider.cpp
===================================================================
--- firefox-trunk-17.0~a1~hg20120807r101567.orig/xpcom/io/nsAppFileLocationProvider.cpp	2012-08-07 03:13:18.000000000 +0100
+++ firefox-trunk-17.0~a1~hg20120807r101567/xpcom/io/nsAppFileLocationProvider.cpp	2012-08-07 11:38:13.751730203 +0100
@@ -60,6 +60,7 @@
 
 #define DEFAULTS_DIR_NAME           NS_LITERAL_CSTRING("defaults")
 #define DEFAULTS_PREF_DIR_NAME      NS_LITERAL_CSTRING("pref")
+#define DEFAULTS_SYSPREF_DIR_NAME   NS_LITERAL_CSTRING("syspref")
 #define DEFAULTS_PROFILE_DIR_NAME   NS_LITERAL_CSTRING("profile")
 #define RES_DIR_NAME                NS_LITERAL_CSTRING("res")
 #define CHROME_DIR_NAME             NS_LITERAL_CSTRING("chrome")
@@ -124,6 +125,15 @@
                 rv = localFile->AppendRelativeNativePath(DEFAULTS_PREF_DIR_NAME);
         }
     }
+    else if (nsCRT::strcmp(prop, NS_APP_SYSPREF_DEFAULTS_50_DIR) == 0)
+    {
+        rv = CloneMozBinDirectory(getter_AddRefs(localFile));
+        if (NS_SUCCEEDED(rv)) {
+            rv = localFile->AppendRelativeNativePath(DEFAULTS_DIR_NAME);
+            if (NS_SUCCEEDED(rv))
+                rv = localFile->AppendRelativeNativePath(DEFAULTS_SYSPREF_DIR_NAME);
+        }
+    }
     else if (nsCRT::strcmp(prop, NS_APP_PROFILE_DEFAULTS_50_DIR) == 0 ||
              nsCRT::strcmp(prop, NS_APP_PROFILE_DEFAULTS_NLOC_50_DIR) == 0)
     {
Index: firefox-trunk-17.0~a1~hg20120807r101567/modules/libpref/src/Preferences.cpp
===================================================================
--- firefox-trunk-17.0~a1~hg20120807r101567.orig/modules/libpref/src/Preferences.cpp	2012-08-07 03:11:15.000000000 +0100
+++ firefox-trunk-17.0~a1~hg20120807r101567/modules/libpref/src/Preferences.cpp	2012-08-07 11:38:13.755730205 +0100
@@ -1057,6 +1057,15 @@
   if (NS_FAILED(rv))
     NS_WARNING("Error parsing application default preferences.");
 
+
+  /* Load $gre/defaults/syspref/*.js */
+  rv = NS_GetSpecialDirectory(NS_APP_SYSPREF_DEFAULTS_50_DIR, getter_AddRefs(defaultPrefDir));
+  NS_ENSURE_SUCCESS(rv, rv);
+
+  rv = pref_LoadPrefsInDir(defaultPrefDir, specialFiles, NS_ARRAY_LENGTH(specialFiles));
+  if (NS_FAILED(rv))
+    NS_WARNING("Error parsing application system preferences.");
+
   // Load jar:$app/omni.jar!/defaults/preferences/*.js
   // or jar:$gre/omni.jar!/defaults/preferences/*.js.
   nsRefPtr<nsZipArchive> appJarReader = mozilla::Omnijar::GetReader(mozilla::Omnijar::APP);
